graphql-java-servlet/src/main/java/graphql/kickstart/servlet/QueryResponseWriter.java

```diff
@@ -2,28 +2,42 @@
 
 import graphql.kickstart.execution.GraphQLQueryResult;
 import graphql.kickstart.execution.GraphQLObjectMapper;
+import graphql.kickstart.execution.input.GraphQLInvocationInput;
+import graphql.kickstart.servlet.cache.CachingQueryResponseWriter;
+import graphql.kickstart.servlet.cache.GraphQLResponseCacheManager;
+
 import java.io.IOException;
 import java.util.Objects;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-interface QueryResponseWriter {
+public interface QueryResponseWriter {
 
   static QueryResponseWriter createWriter(
```

Comment:
I'd leave this method as is and instead add another static method to extend the existing one, e.g.
```java
static QueryResponseWriter createCacheWriter(...) {
  QueryResponseWriter writer = createWriter(...); // call the other static method here
  if (responseCache != null) {
    return new CachingQueryResponseWriter(writer, responseCache, invocationInput, result.isError());
  }
  return writer;
}
```
It would probably be even better though if this static factory method is replaced with the abstract factory pattern moving the caching logic further away from the core logic.

Reply:
Done. Pls check post code review fixes.